io.mrarm.mcpelauncher-ui.json

```diff
@@ -0,0 +1,128 @@
+    "base": "io.qt.qtwebengine.BaseApp",
+    "base-version": "5.14",
+    "command": "mcpelauncher-ui-qt",
+    "build-options" : {
+        "cflags": "-O2 -g -DNDEBUG",
+        "cxxflags": "-O2 -g -DNDEBUG"
+    },
+    "cleanup": [
+        "*.a",
+        "*.la",
+        "/bin/protoc",
+        "/bin/zipcmp",
+        "/bin/zipmerge",
+        "/bin/ziptool",
+        "/lib/libprotobuf-lite*",
+        "/lib/libprotoc*"
+    ],
+    "finish-args": [
+        "--socket=x11",
+        "--share=network",
+        "--filesystem=host",
+        "--device=dri",
+        "--talk-name=org.freedesktop.Flatpak"
+     ],
+    "modules": [
+        {
+            "name": "protobuf",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/protocolbuffers/protobuf.git",
+                    "branch": "3.12.x"
+                }
+            ]
+        },
+        {
+            "name": "libzip",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/nih-at/libzip.git",
+                    "branch": "master"
+                }
+            ],
+            "buildsystem": "cmake-ninja"
+        },
+        {
+            "name": "libuv",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/libuv/libuv.git",
+                    "branch": "v1.x"
+                }
+            ],
+            "buildsystem": "cmake-ninja"
+        },
+        {
+            "name": "mcpelauncher-ui",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/minecraft-linux/mcpelauncher-ui-manifest.git",
+                    "branch": "ng"
+                }
+            ],
+            "buildsystem": "cmake-ninja"
+        },
+        {
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/minecraft-linux/mcpelauncher-proprietary.git",
+                    "branch": "master"
+                },
+                {
+                    "type": "file",
+                    "path": "io.mrarm.mcpelauncher-ui.metainfo.xml"
+                }
+            ],
+            "name": "compat",
+            "buildsystem": "simple",
+            "build-commands": [
+                "mkdir -p /app/bin/",
+                "echo '#!/bin/bash' > /app/bin/mcpelauncher-client",
+                "echo 'flatpak-spawn --host flatpak --filesystem=~/.var/app/$FLATPAK_ID:ro run app/io.mrarm.mcpelauncher-client// \"$@\"' >> /app/bin/mcpelauncher-client",
+                "chmod +x /app/bin/mcpelauncher-client",
+                "cd /app/share/applications && mv mcpelauncher-ui-qt.desktop io.mrarm.mcpelauncher-ui.desktop && sed -i 's/Icon=mcpelauncher-ui-qt/Icon=io.mrarm.mcpelauncher-ui/g' io.mrarm.mcpelauncher-ui.desktop",
```

Comment:
These should be installed under correct names by your build system instead of renamed and sed'ed here.

Reply:
flatpak builder wants the applicationid be the filename, but the mcpelauncher-ui-qt submodule has it's own application id.
mcpelauncher-ui-qt isn't an valid flatpak name, but valid for all other builds.
Now added the right flatpak desktop file into the repo without using sed or renaming.